core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala

```diff
@@ -908,6 +909,44 @@ class GroupMetadataManagerTest {
     assertTrue(group.has(memberId))
   }
 
+  @Test
+  def testShouldThrowExceptionForUnsupportedGroupMetadataVersion(): Unit = {
+    val generation = 1
+    val protocol = "range"
+    val memberId = "memberId"
+    val unsupportedVersion = Short.MinValue
+
+    // put the unsupported version as the version value
+    val groupMetadataRecordValue = buildStableGroupRecordWithMember(generation, protocolType, protocol, memberId)
+      .value().putShort(unsupportedVersion)
+    // reset the position to the starting position 0 so that it can read the data in correct order
+    groupMetadataRecordValue.position(0)
+
+    val e = assertThrows(classOf[KafkaException],
+      () => GroupMetadataManager.readGroupMessageValue(groupId, groupMetadataRecordValue, time))
```

Comment:
This is throwing an error:

```
10:23:54  /home/jenkins/workspace/confluentinc-pr_kafka_PR-432/core/src/test/scala/unit/kafka/coordinator/group/GroupMetadataManagerTest.scala:925: overloaded method value assertThrows with alternatives:
10:23:54    [T <: Throwable](x$1: String, x$2: Class[T], x$3: org.junit.function.ThrowingRunnable)T <and>
10:23:54    [T <: Throwable](x$1: Class[T], x$2: org.junit.function.ThrowingRunnable)T
10:23:54   cannot be applied to (Class[org.apache.kafka.common.KafkaException], () => kafka.coordinator.group.GroupMetadata)
10:23:54      val e = assertThrows(classOf[KafkaException],
10:23:54              ^
```

Reply:
Cherry-picked a fix from https://github.com/apache/kafka/commit/63f3e1c33b03870c5d4fe2dba40cf385cfab6dca for this test.